core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -806,6 +806,22 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
     }
     // Clean the blacklist from the expired entries.
     clearBlacklist(CLEAN_INTERVAL_S)
+
+    // Delete driver logs from the configured spark dfs dir that exceed the configured max age
+    try {
+      val hdfsDir = conf.get("spark.driver.log.dfsDir")
+      val appDirs = fs.listLocatedStatus(new Path(hdfsDir))
+      while (appDirs.hasNext()) {
+        val appDirStatus = appDirs.next()
+        if (appDirStatus.getModificationTime() < maxTime) {
+          logInfo(s"Deleting expired driver log for: ${appDirStatus.getPath().getName()}")
+          deleteLog(appDirStatus.getPath())
```

Comment:
I added the documentation but I am still relying on the existing history server configurations. This is because I am using an existing cleaner. Otherwise, I may have to create another cleaner for this, which can run at a different interval and has a different configured max age.

Please let me know if I should create another cleaner for this or just provide a configuration for a different max age.

Reply:
It's pretty trivial to create a different task to run this code you're adding...